src/derivative_operators/abstract_operator_functions.jl

```diff
@@ -0,0 +1,211 @@
+# ~ bound checking functions ~
+checkbounds(A::AbstractDerivativeOperator, k::Integer, j::Integer) =
+    (0 < k ≤ size(A, 1) && 0 < j ≤ size(A, 2) || throw(BoundsError(A, (k,j))))
+
+checkbounds(A::AbstractDerivativeOperator, kr::Range, j::Integer) =
+    (checkbounds(A, first(kr), j); checkbounds(A,  last(kr), j))
+
+checkbounds(A::AbstractDerivativeOperator, k::Integer, jr::Range) =
+    (checkbounds(A, k, first(jr)); checkbounds(A, k,  last(jr)))
+
+checkbounds(A::AbstractDerivativeOperator, kr::Range, jr::Range) =
+    (checkbounds(A, kr, first(jr)); checkbounds(A, kr,  last(jr)))
+
+checkbounds(A::AbstractDerivativeOperator, k::Colon, j::Integer) =
+    (0 < j ≤ size(A, 2) || throw(BoundsError(A, (size(A,1),j))))
+
+checkbounds(A::AbstractDerivativeOperator, k::Integer, j::Colon) =
+    (0 < k ≤ size(A, 1) || throw(BoundsError(A, (k,size(A,2)))))
+
+
+# BandedMatrix{A,B,C,D}(A::DerivativeOperator{A,B,C,D}) = BandedMatrix(full(A, A.stencil_length), A.stencil_length, div(A.stencil_length,2), div(A.stencil_length,2))
+
+# ~~ getindex ~~
+@inline function getindex(A::AbstractDerivativeOperator, i::Int, j::Int)
+    @boundscheck checkbounds(A, i, j)
+    mid = div(A.stencil_length, 2) + 1
+    bpc = A.stencil_length - mid
+    l = max(1, low(j, mid, bpc))
+    h = min(A.stencil_length, high(j, mid, bpc, A.stencil_length, A.dimension))
+    slen = h - l + 1
+    if abs(i - j) > div(slen, 2)
+        return 0
+    else
+        return A.stencil_coefs[mid + j - i]
```

Comment:
Looks like for now this just does the stencil matrix without boundary conditions?